codepropertygraph/src/main/scala/io/shiftleft/passes/ParallelCpgPass.scala

```diff
@@ -33,7 +36,8 @@ abstract class ParallelCpgPass[T](cpg: Cpg, outName: String = "", keyPools: Opti
                             prefix: String = "",
                             inverse: Boolean = false)(f: Writer => Unit): Unit = {
     val writer = new Writer(serializedCpg, prefix, inverse)
-    val writerThread = new Thread(writer)
+    val mdc = MDC.getCopyOfContextMap
+    val writerThread = new Thread({ MDC.setContextMap(mdc); writer })
```

Comment:
Can we wrap this in a `createThread` method?

Reply:
We could. But this logic is needed in 2 places (one of which is legacy and due for deprecation) in one file and is < 1 extra line.

Therefore it makes no sense to abstract this.

Reply:
I moved the logic around so that both Writer classes use the same.